### DISCUSSION TITLE
Separate grouping for dependency dashboard

### DISCUSSION THREAD

**Discussion by fxshlein** [renovatebot/renovate#31167] [Suggest an Idea]
### Tell us more.

I was just configuring renovate, and my idea was to apply the following logic:
- Enable dependency dashboard approval for all dependencies
- Extend from `group:all` so all dependencies are grouped together in a single PR

My goal with this was that you could individually select dependencies from the list in the dashboard, and then renovate would create a single PR with the selected dependencies. I then found out (and it makes sense in hindsight) that this can't really work, because they just show up in the dashboard as a single group that you can either check or uncheck.

Perhaps this could be a cool feature tho, because conceptually, there is a difference between a group like `... monorepo` and `all non-major dependencies`. The first is an actual set of artifacts that has to be updated together, the second is more of a category. Maybe it could be useful to somehow split these two concepts:
- The existing `groupName`: Specifies the name for a set of packages that actually belong together.
- A new `categoryName`: Specifies a category to group pull requests.

Then, PRs would be created for each category, but the dashboard would show individual groups.

This could also nicely interact with the various proposals to keep versions for dependencies that belong together aligned: If the global mechanisms like `all non-major dependencies` work with categories instead of groups, then the version alignment can happen on the group level.

For backwards compatibility with custom configurations, it could be an idea that the `categoryName` simply falls back to the `groupName` if it's not set.

**Comment by Liquidmasl** [comment]
We have a monorepo and it would be awesome if we could categories the updates in the dashboard. As something seperate to groups. 

Currently the dashboard is hard to use without that. 
I helped myself prefitting those subsystems to the commit message. but its suboptimal.

something like..

## Pending Approval

These branches will be created by Renovate only once you click their checkbox below.

### Backbone
 - [ ] Update dependency AWSSDK.S3 to 3.7.419.1
 - [ ] Update dependency Hangfire.Mongo to 1.11.6
 - [ ] Update dependency Microsoft.VisualStudio.Azure.Containers.Tools.Targets to 1.21.2
 - [ ] Update docker dependencies (`mongo`, `nginx`, `node`, `python`)
 - [ ] Update dotnet monorepo to 8.0.17 (`Microsoft.AspNetCore.Authentication.JwtBearer`, `Microsoft.AspNetCore.Mvc.Testing`)

### Frontend
 - [ ] Update Hangfire monorepo to 1.8.20 (`Hangfire.AspNetCore`, `Hangfire.Core`)
 - [ ] Update dependency FluentAssertions to 6.12.2
 - [ ] Update dependency Moq to 4.20.72


### Some service
...



# Proposal

I would like to just manually be able to set a category. this is just for visibility in the dashboard. 

something like 

```json
    {
      "matchFileNames": [
        "modules/sub_module/**"
      ],
      "dashboardCategory": "sub_module",
    }
```

So this does not interfere with branchname, commit message, pr title, groups, etc. 

## Use as labels
Additionally, optinally, these categorys could be added as labels to the PRs

**Reply by astellingwerf** [reply]
What do you think? Does it mean there should be check boxes like "Create all pending approval PRs at once" and "Create all rate-limited PRs at once" per group?

**Reply by astellingwerf** [reply]
In my first attempt to implement this, I've decided to put the checkbox under an extra header "All", like so:
<img width="878" alt="Screenshot 2025-06-29 at 17 56 26" src="https://github.com/user-attachments/assets/f4433481-6361-4c16-b6aa-a73439f1dc6b" />
See https://github.com/astellingwerf/renovate-dependency-dashboard-categories/issues/3

**Reply by 3bbbeau** [reply]
From an end user perspective, this is really great. 

We use renovate to manage cluster dependencies, and being able to sort them like this would be absolutely fantastic as there are components that overlap.

**Reply by Liquidmasl** [reply]
that looks awesome!
Wow I didnt expect anyone picking up on that suggestion this fast!

This will make the dashboard so much more manageable!
